gnovm/tests/files/const53.gno

```diff
@@ -0,0 +1,13 @@
+package main
+
+import "fmt"
+
+func main() {
+	const a = len([1]int{
+		func() int { println("hey!!"); return 1 }(),
+	})
+	fmt.Println("Hello", a)
+}
+
+// Error:
```

Comment:
we can also add another test to clarify, which is valid:
```go
nestedArr := [1]int{func() int { println("hey!!"); return 1 }()}
const a = len(nestedArr)
```

Reply:
should be done here https://github.com/gnolang/gno/pull/3714/commits/8c71cc67909250237884cb7bf4bbaf2050c7e2b3 ^^